docs/workflows/executions/manual-partial-and-production-executions.md

```diff
@@ -0,0 +1,54 @@
+---
+#https://www.notion.so/n8n/Frontmatter-432c2b8dff1f43d4b1c8d20075510fe4
+title: Manual, partial, and production executions
+description: How manual, partial, and automatic workflow executions differ.
+contentType: explanation
+---
+
+# Manual, partial, and production executions
+
+There are some important differences in how n8n executes workflows manually (by clicking the **Test Workflow** button) and automatically (when the workflow is **Active** and triggered by an event or schedule).
+
+## Manual executions
+
+Manual executions allow you to run workflows directly from the [canvas](/glossary.md#canvas-n8n) to test your workflow logic. These executions are "ad-hoc" — they run only when you manually select the **Test workflow** button.
+
+Manual executions make building workflows easier by allowing you to iteratively test as you go, following the flow logic and seeing data transformations. You can test conditional branching, data formatting changes, and loop behavior by providing different input items and modifying node options.
+
+/// note | Pinning execution data
+When performing manual executions, you can use [data pinning](/data/data-pinning.md) to "pin" or "freeze" the output data of a node. On future runs, instead of executing the pinned node, n8n will substitute the pinned data and continue following the flow logic. This allows you to iterate without operating on variable data or repeating queries to external services. Production executions ignore all pinned data.
+///
+
+## Partial executions
+
+Clicking the **Test workflow** button at the bottom of the workflow in the **Editor** tab manually runs the entire workflow. You can also perform partial executions to run specific steps in your workflow. Partial executions are manual executions that only run a subset of your workflow nodes.
+
+To perform a partial execution, select a node, open its detail view, and select **Test step**. This executes the specific node and, if required, any preceding nodes required to fill in its input data. You can also temporarily disable specific nodes in the workflow chain to avoid interacting with those services while building.
```

Comment:
too many 'required's - maybe just omit the first?
```suggestion
To perform a partial execution, select a node, open its detail view, and select **Test step**. This executes the specific node and any preceding nodes required to fill in its input data. You can also temporarily disable specific nodes in the workflow chain to avoid interacting with those services while building.
```

Reply:
Oooh, good catch :+1: